staging/src/k8s.io/kube-aggregator/pkg/apiserver/handler_proxy.go

```diff
@@ -248,37 +244,38 @@ func (r *proxyHandler) updateAPIService(apiService *apiregistrationv1api.APIServ
 		return
 	}
 
-	newInfo := proxyHandlingInfo{
-		name: apiService.Name,
-		restConfig: &restclient.Config{
-			TLSClientConfig: restclient.TLSClientConfig{
-				Insecure:   apiService.Spec.InsecureSkipTLSVerify,
-				ServerName: apiService.Spec.Service.Name + "." + apiService.Spec.Service.Namespace + ".svc",
-				CertData:   r.proxyClientCert,
-				KeyData:    r.proxyClientKey,
-				CAData:     apiService.Spec.CABundle,
+	transportConfig := &transport.Config{
+		TLS: transport.TLSConfig{
+			Insecure: apiService.Spec.InsecureSkipTLSVerify,
+			GetCert: func() (*tls.Certificate, error) {
+				cert, key := r.certKeyContentProvider.CurrentCertKeyContent()
+				crt, err := tls.X509KeyPair(cert, key)
+				return &crt, err
```

Comment:
okay - even on `err`, `crt` will not be nil - though it could change in the future releases.

Reply:
so ... should I rather return nil here?